src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -104,6 +124,67 @@ public static ImmutableArray<ILocalSymbol> GetDeclaredVariables(this IVariableDe
 
             return arrayBuilder.ToImmutableAndFree();
         }
+
+        /// <summary>
+        /// Get an optional argument name for a named argument to the given <paramref name="dynamicExpression"/> at the given <paramref name="index"/>.
+        /// </summary>
+        /// <param name="dynamicExpression">Dynamic or late bound expression.</param>
+        /// <param name="index">Argument index.</param>
+        public static string GetArgumentName(this IHasDynamicArgumentsExpression dynamicExpression, int index)
+        {
+            if (dynamicExpression == null)
+            {
+                throw new ArgumentNullException(nameof(dynamicExpression));
+            }
+
+            var expression = (HasDynamicArgumentsExpression)dynamicExpression;
+            if (expression.ArgumentNames.IsDefaultOrEmpty)
+            {
+                return null;
+            }
+
+            if (index < 0 || index >= expression.ArgumentNames.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
```

Comment:
This check is unnecessary, `ArgumentNames[index]` will throw the same exception.